File: Forge/src/main/java/hantonik/atomic/core/util/helper/ACConfigHelper.java
package hantonik.atomic.core.util.helper;

import com.electronwill.nightconfig.core.file.CommentedFileConfig;
import com.electronwill.nightconfig.core.io.WritingMode;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraftforge.common.ForgeConfigSpec;
import net.minecraftforge.fml.loading.FMLPaths;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class ACConfigHelper {
    public static void load(ForgeConfigSpec config, String location) {
        var path = FMLPaths.CONFIGDIR.get().resolve(location);
        var data = CommentedFileConfig.builder(path).sync().autosave().writingMode(WritingMode.REPLACE).build();
        data.load();

        config.setConfig(data);
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/util/helper/ACConfigHelper.kt
package hantonik.atomic.core.util.helper

import com.electronwill.nightconfig.core.file.CommentedFileConfig
import com.electronwill.nightconfig.core.io.WritingMode
import net.minecraftforge.common.ForgeConfigSpec
import net.minecraftforge.fml.loading.FMLPaths

object ACConfigHelper {
    fun load(config: ForgeConfigSpec, location: String) {
        val path = FMLPaths.CONFIGDIR.get().resolve(location)
        val data = CommentedFileConfig.builder(path).sync().autosave().writingMode(WritingMode.REPLACE).build()
        data.load()

        config.setConfig(data)
    }
}